holoviews/core/operation.py

```diff
@@ -214,6 +215,10 @@ def __call__(self, element, **kwargs):
                 return self._apply(element)
         elif 'streams' not in kwargs:
             kwargs['streams'] = self.p.streams
+
+
+        if isinstance(kwargs['streams'], dict):
+            kwargs['streams'] = streams.streams_list_from_dict(kwargs['streams'])
```

Comment:
I think we want something else here tbh, I'll take a look at this once you're done.

Reply:
Sure. I would also like to push this logic deeper into `Apply` and `Dynamic` so it can all be more generally applicable. This version was all about seeing is this whole idea is at all feasible...